lib/logqueue-fifo.c

```diff
@@ -267,6 +268,7 @@ log_queue_fifo_push_tail(LogQueue *s, LogMessage *msg, const LogPathOptions *pat
            * input thread finishes */
 
           main_loop_worker_register_batch_callback(&self->qoverflow_input[thread_id].cb);
+          self->qoverflow_input[thread_id].cb.user_data = log_queue_ref(&self->super);
           self->qoverflow_input[thread_id].finish_cb_registered = TRUE;
```

Comment:
Together with the other note further down, I think we should not change user data here, we should just take a ref, and before registering the callback.

I would also document why that ref is needed, probably where the rest of the locking assumptions are documented.